R/coveralls.R

```diff
@@ -57,9 +71,45 @@ to_coveralls <- function(x, service_job_id = Sys.getenv("TRAVIS_JOB_ID"),
       "source_files" = res)
   }
 
+  if (!is.null(git_info)) payload <- c(payload, git = list(git_info))
+
   jsonlite::toJSON(na = "null", payload)
 }
 
+jenkins_git_info <- function() {
+  # check https://coveralls.zendesk.com/hc/en-us/articles/201350799-API-Reference
+  # for why and how we are doing this
+  formats <- c(
+    id = "%H",
+    author_name = "%an",
+    author_email = "%ae",
+    commiter_name = "%cn",
+    commiter_email = "%ce",
+    message = "%s"
+  )
+  head <- lapply(structure(
+    scan(
+      sep="\030", # http://en.wikipedia.org/wiki/Delimiter#ASCII_delimited_text
+      what = "character",
+      text=system(intern=TRUE,
+        paste0("git log -n 1 --pretty=format:",
+          paste(collapse="%n", formats)
```

Comment:
no the collapse argument.  If scan is expecting a separator of `\030` you need to have git log put a `\030` between each format as well i.e. `paste(collapse="\030"`

Right now you are outputting a newline separator from git log but trying to read using a ascii record separator.